index.html

```diff
@@ -4054,51 +4058,89 @@ <h2>Elements</h2>
- known elements</dfn>.
- When the <a>browsing context</a> is <a>discarded</a>,
- the <a>list of known elements</a> is discarded along with it.
+<p>The <dfn>WebDriver node id</dfn> is a globally unique string
+representing a handle to a DOM node in a specific
+WebDriver <a>session</a>.
+
+<p>A WebDriver <a>session</a> has a <dfn>browsing context group node
+map</dfn>, which is a weak map between a <a>browsing context group</a>
+and a <a>node id map</a>.
 
-<p>To <dfn>get a known connected element</dfn> with
-argument <var>reference</var>, run the following steps:
+<p>A <dfn>node id map</dfn> is weak map between nodes and their
+corresponding <a>WebDriver node id</a>.
+
+<p>To <dfn>get a node</dfn> given <var>session</var>
+and <var>reference</var>:
 <ol>
-  <li>Let <var>element</var> be the item in the <a>current browsing
-  context</a>’s <a>list of known elements</a> for which the <a>web
-  element reference</a> is equal to <var>reference</var>, if such an
-  element exists. Otherwise return <a>error</a> with <a>error
-  code</a> <a>no such element</a>.
-  <li>If <var>element</var> <a>is stale</a>, return <a>error</a>
-  with <a>error code</a> <a>stale element reference</a>.
-  <li>Return <a>success</a> with <var>element</var>.
+  <li>Let <var>browsing context group node map</var>
+  be <var>session</var>'s <a>browsing context group node map</a>.
+
+  <li>Let <var>browsing context group</var> be <a>current browsing
+  context</a>'s <a>browsing context group</a>.
+
+  <li>If <var>browsing context group node map</var> does not
+  contain <var>browsing context group</var>, return null.
+
+  <li>Let <var>node id map</var> be <var>browsing context group node
+  map</var>[<var>browsing context group</var>].
+
+  <li>Let <a>node</a> be the entry in <var>node id map</var> whose
+  value is <var>reference</var>, if such an entry exists, or null
+  otherwise.
+
+  <li>Return <var>node</var>.
 </ol>
 
-<p>To <dfn>get or create a web element reference</dfn>
- with <a><var>element</var></a>:
+<p>To <dfn>get or create a node reference</dfn>
+given <var>session</var> and <var>node</var>:
+<ol>
+  <li>Let <var>browsing context group node map</var>
+  be <var>session</var>'s <a>browsing context group node map</a>.
+
+  <li>Let <var>browsing context group</var> be <a>current browsing
+  context</a>'s <a>browsing context group</a>.
 
+  <li>If <var>browsing context group node map</var> does not
+  contain <var>browsing context group</var>, set <var>browsing context
+  group node map</var>[<var>browsing context group</var>] to a new weak map.
+
+  <li>Let <var>node id map</var> be<var>browsing context group node
+  map</var>[<var>browsing context group</var>].
+
+  <li>If <var>node id map</var> does not contain <var>node</var>,
+  set <var>node id map</var>[<var>node</var>] to a new globally
+  unique string.
+
+ <li><p>Return <var>node id map</var>[<var>node</var>].
+</ol>
+
+<p>To <dfn>get a known element</dfn> given <var>reference</var>:
```

Comment:
What is the reason in renaming the method? It still "get a known connected element", as it checks if the element is connected via `If node is stale ...`. Just to unify with "get a known shadow root"?

Reply:
Yeah, it just seemed preferable to go with the shorter name given that the semantics are more like "get a known element connected to the document in the current browsing context", and we don't say any of that for shadow root.